x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/ingest/GeoGridProcessor.java

```diff
@@ -0,0 +1,367 @@
+        }
+
+        @Override
+        List<String> makeNonChildren() {
+            return asStringList(H3.h3ToNoChildrenIntersecting(h3));
+        }
+
+        @Override
+        int getDepth() {
+            return H3.getResolution(h3);
+        }
+
+        private static Geometry makePolygonFromH3(long h3) {
+            return H3CartesianUtil.getNormalizeGeometry(h3);
+        }
+
+        private static List<String> asStringList(long[] cells) {
+            ArrayList<String> addresses = new ArrayList<>(cells.length);
+            for (long cell : cells) {
+                addresses.add(H3.h3ToString(cell));
+            }
+            return addresses;
+        }
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    String field(String key) {
+        return config.field(key);
+    }
+
+    GeometryParserFormat targetFormat() {
+        return targetFormat;
+    }
+
+    TileFieldType tileType() {
+        return tileFieldType;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        public GeoGridProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+        ) {
+            String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
+            String targetField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "target_field", field);
+            String parentField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "parent_field", "");
+            String childrenField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "children_field", "");
+            String nonChildrenField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "non_children_field", "");
+            String depthField = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "depth_field", "");
+            boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "ignore_missing", false);
+            TileFieldType tileFieldType = TileFieldType.parse(
+                ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "tile_type")
+            );
+            GeometryParserFormat targetFormat = getTargetFormat(processorTag, config);
+            FieldConfig fields = new FieldConfig(field, targetField, parentField, childrenField, nonChildrenField, depthField);
+            return new GeoGridProcessor(processorTag, description, fields, ignoreMissing, targetFormat, tileFieldType);
+        }
+
+        private GeometryParserFormat getTargetFormat(String processorTag, Map<String, Object> config) {
+            String propertyName = "target_format";
+            String targetFormat = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, propertyName, "WKT");
```

Comment:
I still think the default type should be GeoJson?

Reply:
Done.